doc/release_checklist.md

```diff
@@ -0,0 +1,26 @@
+Release checklist:
+ * [ ] Write release notes in NEWS. Get them reviewed and merged.
+ * [ ] Ensure your local copy is up to date with master.
+ * [ ] Ensure you can sign commits and that any yubikeys/smartcards you
+       need for that are plugged in.
+ * [ ] Run `./release.sh vX.Y.Z` which will:
+   * Verify the version you specified is a valid semver
+   * Ensure your working directory is clean
+   * Run tests
+   * Tag the release and sign the tag
+   * Verify the the tag
+   * Build release artifacts
+ * [ ] Push the tag to Github with:
+ ```
+ git push vX.Y.Z origin
+ ```
+ * [ ] Sign the release artifacts by running
+ ```
+ gpg -u gpg --local-user 0xCDDE268EBB729EC7 --detach-sign --armor <path to artifact>
+ ```
+   for each release artifact. Do _not_ try to sign all of them at once by globbing. If you do, gpg will
+   sign the combination of all the release artifacts instead of each one individually.
+ * [ ] Create a draft release on Github for the tag and upload all the release artifacts and
+       their signatures. Copy and paste the release notes from NEWS here as well.
+ * [ ] Get someone to review the draft release
```

Comment:
There should be an extra step here of "tag the resulting docker image at `quay.io/coreos/ct` with the new version.", so something like:
```
docker pull quay.io/coreos/ct:v0.6.1-11-g90d21c6
docker tag quay.io/coreos/ct:v0.6.1-11-g90d21c6 quay.io/coreos/ct:v0.7.0
docker push quay.io/coreos/ct:v0.7.0
```

CI for the release PR runs before the git tag exists, so it won't be tagged with the new version.

Reply:
I just did this for v0.7.0, so the instructions should read:

> Once the new version of ct has been pushed to the docker hub, run (versions and hashes will be different):
> ```
> docker pull quay.io/coreos/ct:v0.7.0-2-gac076a8
> docker tag quay.io/coreos/ct:v0.7.0-2-gac076a8 quay.io/coreos/ct:v0.7.0
> docker push quay.io/coreos/ct:v0.7.0
> ```